frontend/src/metabase-lib/queries/StructuredQuery.ts

```diff
@@ -1272,6 +1297,8 @@ class StructuredQueryInner extends AtomicQuery {
       // de-duplicate explicit and implicit joined tables
       const explicitJoins = this._getExplicitJoinsSet(joins);
 
+      // This is only a refactor, I moved this line down to limit the scope where this variable could be used.
+      const dimensionIsFKReference = dimension => dimension.field?.().isFK();
```

Comment:
This is only a small refactor. I don't see the point of declaring `dimensionIsFKReference` too early since it's only been used once on the next line.